lib/core/src/lib/dynamic-chip-list/dynamic-chip-list.component.ts

```diff
@@ -0,0 +1,218 @@
+    @Input()
+    limitChipsDisplayed = false;
+
+    /** Emitted when button for view more is clicked. */
+    @Output()
+    displayNext = new EventEmitter<void>();
+
+    /** Emitted when any chip is removed. */
+    @Output()
+    removedChip = new EventEmitter<string>();
+
+    @ViewChild('nodeListContainer')
+    containerView: ElementRef;
+
+    @ViewChildren(MatChip)
+    matChips: QueryList<MatChip>;
+
+    chipsToDisplay: Chip[] = [];
+    calculationsDone = false;
+    columnFlexDirection = false;
+    moveLoadMoreButtonToNextRow = false;
+    undisplayedChipsCount = 0;
+    viewMoreButtonLeftOffset: number;
+    viewMoreButtonTop = 0;
+    paginationData: Pagination;
+
+    private initialChips: Chip[] = [];
+    private onDestroy$ = new Subject<boolean>();
+    private initialLimitChipsDisplayed: boolean;
+    private viewMoreButtonLeftOffsetBeforeFlexDirection: number;
+    private requestedDisplayingAllChips = false;
+    private resizeObserver = new ResizeObserver(() => {
+        this.calculateChipsToDisplay();
+        this.changeDetectorRef.detectChanges();
+    });
+
+    constructor(private changeDetectorRef: ChangeDetectorRef) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.pagination) {
+            this.limitChipsDisplayed = this.pagination?.hasMoreItems;
+            this.paginationData = this.pagination;
+            this.initialLimitChipsDisplayed = this.limitChipsDisplayed;
+        }
+        if (changes.chips) {
+            this.initialChips = this.chips;
+            this.chipsToDisplay = this.initialChips;
+            if (this.limitChipsDisplayed && this.chipsToDisplay.length) {
+                setTimeout(() => {
+                    this.calculateChipsToDisplay();
+                    this.changeDetectorRef.detectChanges();
+                });
+            }
+        }
+    }
+
+    ngOnInit(): void {
+        if (this.paginationData) {
+            this.limitChipsDisplayed = this.paginationData.hasMoreItems;
+        }
+        this.initialLimitChipsDisplayed = this.limitChipsDisplayed;
+    }
+
+    ngAfterViewInit(): void {
+        this.resizeObserver.observe(this.containerView.nativeElement);
+    }
+
+    ngOnDestroy(): void {
+        this.onDestroy$.next(true);
+        this.onDestroy$.complete();
+        this.resizeObserver.unobserve(this.containerView.nativeElement);
+    }
+
+    displayNextChips(event: Event): void {
+        event.preventDefault();
+        event.stopPropagation();
+        if (this.paginationData) {
+            this.requestedDisplayingAllChips = !this.paginationData.hasMoreItems;
+        } else {
+            this.limitChipsDisplayed = false;
+            this.requestedDisplayingAllChips = true;
+        }
+        if (this.requestedDisplayingAllChips) {
+            this.resizeObserver.unobserve(this.containerView.nativeElement);
+        }
+        this.displayNext.emit();
+    }
+
+    private calculateChipsToDisplay() {
```

Comment:
maybe add return type void here as you specified it for other functions

Reply:
done